include/odp/api/spec/crypto_types.h

```diff
@@ -986,6 +1015,12 @@ typedef struct odp_crypto_packet_result_t {
 	/** Request completed successfully */
 	odp_bool_t  ok;
 
+	/** Input packet passed to odp_crypo_op_enq() when the operation
+	 *  type of the session is ODP_CRYPTO_OP_TYPE_OOP. In other cases
+	 *  this field does not have a valid value.
+	 */
+	odp_packet_t pkt_in;
+
 	/** Cipher status */
 	odp_crypto_op_status_t cipher_status;
```

Comment:
Should we make it such that this filed & auth_status would only have valid values when crypto operation is not OK? Then we can skip updating the field.

Other option is to have a union of both cipher status & auth_status,

union crypto_result_status {
    struct status {
        uint8_t cipher_status;
        uint8_t auth_status;
    } s;
    uint16_t u16;
};

Reply:
We could, but since that is a backward-incompatible change, we could other changes as well, like deprecating the separate HW-error fields and maybe packing the cipher and auth error statuses tighter (not a union but something that can hold both). But all that is should probably be in a separate PR or at least in a separate set of patches.

Reply:
Agreed. This change increases the size of the structure. So, isn't this already backwards-incompatible change? Anyway, please consider reducing the size of the structure when we work on improving the result mechanism.

Reply:
Merely adding fields in a structure is considered backward compatible in ODP, since it does not break reasonably sensibly written code. The field is never needed by old code that does not know about the OOP operation type. It is similar to adding new optionally-filled fields (with good defaults) in some config structure: well written code does not break, but code that relies on the physical size and/or padding details of the structure can break (but it could then break with a new compiler too).

Reply:
Understood.